src/main/java/com/googleinterns/zoomtube/servlets/LectureServlet.java

```diff
@@ -71,17 +74,41 @@ public void init() throws ServletException {
   // TODO: Check if URL is valid.
   public void doPost(HttpServletRequest request, HttpServletResponse response)
       throws IOException, ServletException {
-    Optional<String> optionalVideoUrl = getParameter(request, LINK_INPUT);
-    Optional<Entity> existingEntity = checkUrlInDatabase(optionalVideoUrl);
+    Optional<String> error = validatePostRequest(request);
+    if (error.isPresent()) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, error.get());
+      return;
+    }
+
+    String lectureName = request.getParameter(PARAM_NAME);
```

Comment:
This should move to right before it is used. 

See: 4.8.2.2 Declared when needed
Local variables are not habitually declared at the start of their containing block or block-like construct. Instead, local variables are declared close to the point they are first used (within reason), to minimize their scope. Local variable declarations typically have initializers, or are initialized immediately after declaration.

https://google.github.io/styleguide/javaguide.html#s4.8.2-variable-declarations

Reply:
Done